### PR TITLE
CA-332890: trigger mpath count when multipath dm device added/removed

### PR DESCRIPTION
**Pull request by MarkSymsCtx**
Signed-off-by: Mark Syms <mark.syms@citrix.com>

### CODE REVIEW

File: udev/65-multipath.rules [hunk 1]
```diff
@@ -8,6 +8,9 @@ ACTION=="remove", ENV{DM_MULTIPATH_DEVICE_PATH}=="1",  GOTO="count_mpath"
 IMPORT{db}="DM_ACTION"
 ACTION=="change", ENV{DM_ACTION}=="PATH_REINSTATED", GOTO="count_mpath"
 ACTION=="change", ENV{DM_ACTION}=="PATH_FAILED", GOTO="count_mpath"
+ACTION=="add", PROGRAM=="/bin/bash -c '/sbin/dmsetup table -j %M -m %m | /bin/grep multipath'", ENV{MULTIPATH}="1"
```
Comment (MarkSymsCtx) [inline udev/65-multipath.rules]: Not sure whether this variable should get a product specific prefix or not.

File: udev/65-multipath.rules [hunk 2]
```diff
@@ -8,6 +8,9 @@ ACTION=="remove", ENV{DM_MULTIPATH_DEVICE_PATH}=="1",  GOTO="count_mpath"
...
+IMPORT{db}="MULTIPATH"
+ACTION=="add|remove", ENV{MULTIPATH}=="1", GOTO="count_mpath"
```
Comment (BenSimsCitrix) [inline udev/65-multipath.rules]: Can lines 6 and 7  be removed now because of this?

Comment (BenSimsCitrix) [reply udev/65-multipath.rules]: Ah sorry i see, that is matching is against another ENV variable.

Comment (MarkSymsCtx) [reply udev/65-multipath.rules]: Yes, add/remove DM_MULTPATH_DEVICE_PATH is adding a sub device of the multipath

File: udev/65-multipath.rules (hunk 1 above)
Comment (BenSimsCitrix) [reply udev/65-multipath.rules]: It might be wise to avoid any name collisions.

Comment (BenSimsCitrix) [inline udev/65-multipath.rules]: Do we care about the $? return of this PROGRAM? according to udev man page the stdout goes to RESULT but the key gets set to true if the program succeeds (I assume this means  $? == 0)  which will always be true as running "bash -c" swallows the $?

I think running under bash -ce will propagate the $?

**Pull request merged**